### PR TITLE
Used the parsed version of the query string

### PR DESCRIPTION
**Pull request by robertpi**
Fixes bug APPSEC-1404 as this means a URL decode is automatically applied so the <script> tag will be seen by the WAF.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Core.cs [hunk 1]
```diff
@@ -35,11 +35,17 @@ internal static partial class HttpRequestExtensions
                 cookiesDic.Add(k, request.Cookies[k]);
             }
 
+            var queryStringDic = new Dictionary<string, List<string>>();
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Core.cs:38]: tiny optimisation
```suggestion
            var queryStringDic = new Dictionary<string, List<string>>(request.Query.Count);
```

File: tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs [hunk 2]
```diff
@@ -33,11 +33,18 @@ internal static partial class HttpRequestExtensions
                 cookiesDic.Add(k, request.Cookies[k].Value);
             }
 
+            var queryDic = new Dictionary<string, string>();
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs:36]:
```suggestion
            var queryDic = new Dictionary<string, string>(request.QueryString.AllKeys.Length);
```

**Review by andrewlock — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

File: tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs [hunk 3]
```diff
@@ -27,17 +27,24 @@ internal static partial class HttpRequestExtensions
...
             foreach (var k in request.Cookies.AllKeys)
             {
                 cookiesDic.Add(k, request.Cookies[k].Value);
             }
 
-            var dict = new Dictionary<string, object>()
+            var queryDic = new Dictionary<string, string>(request.QueryString.AllKeys.Length);
+            foreach (var k in request.QueryString.AllKeys)
+            {
+                var values = request.QueryString[k];
```
Comment (anna-git) [inline tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs:39]: no need to apply url decode on each element here?

Comment (robertpi) [reply tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs:39]: right, there's no need, it gets automatically applied when reading the query string this way.

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #1777 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4dcd6b09e206dc92e0d674b84baec93ee7a1dff) will **not change** line coverage
:heavy_check_mark: Merging #1777 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4dcd6b09e206dc92e0d674b84baec93ee7a1dff) will **not change** branch coverage
:no_entry: Merging #1777 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4dcd6b09e206dc92e0d674b84baec93ee7a1dff) will will **increase** complexity by `8`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4dcd6b09e206dc92e0d674b84baec93ee7a1dff) | #1777       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9265` / `14700` | `9263` / `14709` |          |
| Lines %   | `0.63%`      | `0.63%`      |  `0.00%` :heavy_check_mark:  |
| Branches  | `4122` / `6410` | `4120` / `6418` |          |
| Branches %| `0.64%`      | `0.64%`      |  `0.00%` :heavy_check_mark:  |
| Complexity|   `7562`      | `7570`        |  `8`  :no_entry:    |

View the full report for further details:

* [For master](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75004/artifacts?artifactName=Code%20Coverage%20Report_75004&api-version=6.0&%24format=zip)
* [For this PR #1777](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/74948/artifacts?artifactName=Code%20Coverage%20Report_74948&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f4dcd6b09e206dc92e0d674b84baec93ee7a1dff) | #1777       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `0.63%`      | `0.63%`       |  `0.00%` :heavy_check_mark:  |
| Branches %| `0.64%`      | `0.64%`       |  `0.00%` :heavy_check_mark:  |
| Complexity| `7562`      | `7570`       |  `8` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Logging.LogRateLimiter](https://github.com/DataDog/dd-trace-dotnet/tree/98b7751c1d784cf9c0166e34a2bddaa4b2fca530/tracer/src/Datadog.Trace/Logging/LogRateLimiter.cs) | `0.00%` :heavy_check_mark: | `-0.17%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.RuntimeMetrics.RuntimeMetricsWriter](https://github.com/DataDog/dd-trace-dotnet/tree/98b7751c1d784cf9c0166e34a2bddaa4b2fca530/tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs) | `0.00%` :heavy_check_mark: | `-0.08%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [For master](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75004/artifacts?artifactName=Code%20Coverage%20Report_75004&api-version=6.0&%24format=zip)
* [For this PR #1777](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/74948/artifacts?artifactName=Code%20Coverage%20Report_74948&api-version=6.0&%24format=zip)

**Pull request merged**